src/app/views/discover/communities/communities.ts

```diff
@@ -62,7 +62,7 @@ export default class RouteDiscoverCommunities extends BaseRouteComponent {
 	}
 
 	get showCreateCommunity() {
-		return this.app.user && !!this.app.user.can_create_communities && !this.hasMore;
+		return (!this.app.user || !!this.app.user.can_create_communities) && !this.hasMore;
```

Comment:
Hm, this made it kinda hard to read.
I suggest just doing an early-out user check and then hasMore and can_create_communities.
Something like
```
if (!this.app.user) {
    return false;
}

return !!this.app.user.can_create_communities && !this.hasMore;
```

Reply:
The reason this was changed is that we will be showing the "Add Community" widget for non-users as well, using the `AppAuthRequired` directive. We're checking if we either have no user, or if the user is able to create more communities.

Reply:
Oh I see, that's fine then.